### PR TITLE
Remove bounds on T for Debug, PartialEq and Hash impls

### LINKED ISSUE #12: Remove bounds on `T` for `Debug`, `PartialEq` and `Hash` impls
`RawParts` derives `Debug`, `PartialEq`, and `Hash`. The derive macros from `core` add bounds on `T` even when they are not required, as is the case with the `T` in `RawParts` which is only a pointer and impls these traits regardless of what `T` is.

Remove the derived impls and manually implement these traits.

Such a change warrants a minor version bump.

### PR DESCRIPTION
**Pull request by molinarim**
Resolves #12 .

Replaced derived impls with manual implementations of Debug, PartialEq, and Hash. Added relevant tests as well.

### CODE REVIEW

File: src/lib.rs [hunk 1]
```diff
@@ -64,6 +64,8 @@ extern crate alloc;
 
 use alloc::vec::Vec;
 use core::mem::ManuallyDrop;
+use core::hash::{Hash, Hasher}; 
+use core::fmt;
```
Comment (lopopolo) [inline src/lib.rs:68]: Can you alphabetize these imports?

File: src/lib.rs [hunk 2]
```diff
@@ -93,7 +95,7 @@ use core::mem::ManuallyDrop;
 /// };
 /// assert_eq!(rebuilt, [4294967295, 0, 1]);
 /// ```
-#[derive(Debug, PartialEq, Eq, Hash)]
+#[derive(Eq)]
```
Comment (lopopolo) [inline src/lib.rs:98]: I think clippy will complain when deriving `Eq` when `PartialEq` is implemented manually.

Please remove the `#[derive(Eq)]` and replace it with:

```rust
impl<T> Eq for RawParts<T>
```

Please put this below the `PartialEq` implementation.

File: src/lib.rs [hunk 3]
```diff
@@ -311,4 +340,61 @@ mod tests {
         let raw_parts = RawParts::from_vec(vec);
         assert_eq!(raw_parts.capacity, 100);
     }
+
+    #[test]
```
Comment (lopopolo) [inline src/lib.rs:344]: Thanks for adding the tests!

**Review by lopopolo — CHANGES_REQUESTED** [review]
Thanks for the PR @molinarim. This is looking pretty good.

I think there are a few things clippy and rustfmt will complain about. I'll kick off the builds so you can see the failures. One other way you can test this locally for yourself is to run the following, which will require Ruby to be in your PATH:

```sh
bundle install
bundle exec rake
bundle exec rake test
```

**Comment by lopopolo** [comment]
Once we get the tests passing, @molinarim would you please squash this branch? I noticed some commits had vim `.swp` files checked in.

If you'd like to regenerate the .gitignore in this repo to include vim ignores, I'd accept a followup PR for those changes.

File: src/lib.rs (hunk 1 above)
Comment (molinarim) [reply src/lib.rs:68]: Sure! I must've added those in the order I worked on the impls but I'll alphabetize them instead.

File: src/lib.rs (hunk 2 above)
Comment (molinarim) [reply src/lib.rs:98]: Will do.

**Comment by molinarim** [comment]
Thanks for all the feedback, I'll make the suggested changes and open a new PR.

**Pull request closed without merging**